### DISCUSSION TITLE
CNCF Tag Security Self Assessment Review

### DISCUSSION THREAD

**Discussion by Nihal-Rahman** [fluent/fluentd#4352] [Q&A]
Hello!  My name is Nihal Rahman and I am a senior studying at NYU Tandon School of Engineering. I'm taking a computer security course with Professor Justin Cappos. For our assignment, my group and I were assigned to conduct a self-assessment for the Fluentd project through the [tag security Git Hub ](https://github.com/Rana-KV/tag-security) as part of the CNCF initiative.

For the next part of the assignment, my group and I are required to meet with the maintainers of the Fluentd project. This is so that we can gather feedback on our self-assessment and improve upon on it through the addition and expansion of any of the sections. Our self-assessment can be found [here](https://github.com/Nihal-Rahman/tag-security/blob/main/assessments/projects/fluentd/self-assessment.md). Additionally we would love to schedule a time to meet with you guys to review the assessment if possible. If not any feedback regarding the assessment would be highly appreciated.

**Comment by cosmo0920** [comment]
I read your posted self-assessment. As I mentioned in the DM,

> BTW, at a first glance, I noticed that Cure53 report is mainly convered for fluent-bit instead of fluentd:
https://github.com/Nihal-Rahman/tag-security/blob/main/assessments/projects/fluentd/self-assessment.md#appendix
Currently, Fluent Bit is a sub project of Fluentd.

The Cure53 report is mainly related to Fluent Bit. So, it should be removed from the report.

Also, I read other tag security reports on that repo: https://github.com/cncf/tag-security/tree/main/assessments/projects

Once reading the reports, I thought that the security suggestions are still missing in the report. Or, not required for that report?
Even if this part is not mandatory to the report, but we need to add/review suggestions as maintainers.

e.g.)

* https://github.com/cncf/tag-security/tree/main/assessments/projects/opa#recommendations
* https://github.com/cncf/tag-security/tree/main/assessments/projects/pixie#feedback-to-pixie-team

For example, this is just my opinion -- but, currently, Fluentd is trying to follow supply chain security with using [DCO](https://probot.github.io/apps/dco/).
ref: Page11, https://github.com/cncf/tag-security/blob/main/supply-chain-security/supply-chain-security-paper/CNCF_SSCP_v1.pdf

Also, Fluentd is built on top of Ruby ecosystem. So, the possible recommendation is checking licenses of dependent gems.

Another thing, Fluentd is already obtained the certified badge for following OpenSSF guideline.
https://www.bestpractices.dev/projects/1189#reporting

**Comment by cosmo0920** [comment]
@ashie @kenhys @daipom What do y'all think about this self-assessment report?

**Reply by cosmo0920** [reply]
As discussed with @Nihal-Rahman, it would be great that we provide SBOM files if possible. I already raised an issue for SBOM here: https://github.com/fluent/fluentd-docs-gitbook/issues/481

This shouldn't be one-shot issue but it's for adding security contexts of Fluentd.

**Reply by cosmo0920** [reply]
In Japanese, I found this slide which is described for why SBOM is useful.

The page: https://ict-isac.jp/news/news20230208.html
The direct link: https://ict-isac.jp/news/news20230208/20230303%E8%A5%BF%E9%87%8E%E6%A7%98%E8%B3%87%E6%96%99.pdf

**Comment by cosmo0920** [comment]
@Nihal-Rahman Thank you for your great work. We appreciate for that! :smile:
